packages/devextreme/js/__internal/ui/selection/m_selection.strategy.ts

```diff
@@ -88,8 +88,9 @@ export default class SelectionStrategy {
   }
 
   _loadFilteredData(remoteFilter, localFilter?: any, select?: any, isSelectAll?: boolean) {
+    const keyField = this.options.key();
     const filterLength = encodeURI(JSON.stringify(this._removeTemplateProperty(remoteFilter))).length;
-    const needLoadAllData = this.options.maxFilterLengthInRequest && (filterLength > this.options.maxFilterLengthInRequest);
+    const needLoadAllData = (this.options.maxFilterLengthInRequest && (filterLength > this.options.maxFilterLengthInRequest)) || isFunction(keyField);
```

Comment:
[8af98af](https://github.com/DevExpress/DevExtreme/pull/27659/commits/8af98af7e18d8f135f0d0b670a94574346dbf72c)

[accd01e](https://github.com/DevExpress/DevExtreme/pull/27659/commits/accd01eb772309abdd91fbc0fa37b89864aadb64)

Reply:
new fix - [accd01e](https://github.com/DevExpress/DevExtreme/pull/27659/commits/accd01eb772309abdd91fbc0fa37b89864aadb64)

fixed the issue but failed for this test

![image](https://github.com/DevExpress/DevExtreme/assets/133762589/ebf28a25-2f67-48a4-b174-900e1fe3c70c)

extended the condition even further to avoid failing tests

[030dd92](https://github.com/DevExpress/DevExtreme/pull/27659/commits/030dd927fbb4e35a4b613f3150b130a803969b6c)